static/app/components/tabs/tab.tsx

```diff
@@ -58,46 +57,44 @@ export interface BaseTabProps {
    * `variant='filled'` since other variants do not have a border
    */
   borderStyle?: 'solid' | 'dashed';
+  ref?: React.Ref<HTMLLIElement>;
   to?: string;
   variant?: 'flat' | 'filled' | 'floating';
 }
 
-export function BaseTab({
-  ref: forwardedRef,
-  ...props
-}: BaseTabProps & {
-  ref?: React.Ref<HTMLLIElement>;
-}) {
-  const {
-    to,
-    orientation,
-    overflowing,
-    tabProps,
-    hidden,
-    isSelected,
-    variant = 'flat',
-    borderStyle = 'solid',
-    as = 'li',
-  } = props;
-
-  const ref = useObjectRef(forwardedRef);
-  const InnerWrap = useCallback(
```

Comment:
this is a weird and dangerous pattern; components should be standalone stable components, not inline functions. I’ve extracted this out of the function body into its own component.